sem/utils.py

```diff
@@ -330,3 +331,104 @@ def compute_sensitivity_analysis(
 #                  np.array(xarray.sel(**kwargs)).squeeze())
 #     interact(plot_line, **{k: v for k, v in param_ranges.items() if k != x_axis
 #                            and len(v) > 1})
+
+def parse_log_component(log_component, ns3_log_components=None):
+    """
+    Verifies if the log levels/log classes passed in the log_component
+    dictionary are valid and converts log levels to corresponding log classes.
+    Returns a dictionary with the valid components and log classes.
+    For example,
+    'level_debug' gets converted to 'warn|error|debug'
+
+    Args:
+        log_component (dict): a python dictionary with the
+            log_components (to enable) as the key and the log_levels
+            as the value. Log levels should be written in the same format as
+            the one specified by the ns-3 manual for the environment variable
+            NS_LOG.
+            Note: If any prefix is mentioned, it will be dropped and
+            prefix_all will always be appended.
+
+            For example,
+            log_component = {
+                'component1' : 'info',
+                'component2' : 'level_debug|info'
+            }
+        ns3_log_components (list): A list containing all the valid log
+            components supported by ns-3.
+    """
+    if not log_component:
+        return None
```

Comment:
you mean the following? if so, leave it as it is, otherwise be sure to improve the condition
```suggestion
    if log_component is not None or log_component == {}:
        return None
```